src/main/java/org/kohsuke/github/GitHubClient.java

```diff
@@ -52,10 +53,8 @@
 
     private HttpConnector connector;
 
-    private final Object rateLimitLock = new Object();
-
     @Nonnull
-    private GHRateLimit rateLimit = GHRateLimit.DEFAULT;
+    private final AtomicReference<GHRateLimit> rateLimit = new AtomicReference<>(GHRateLimit.DEFAULT);
```

Comment:
The container is final the value changes.  Is this misleading or okay?

Reply:
fine to me you append to lists that are 'final'.

any reason not to do the same to https://github.com/hub4j/github-api/pull/1134/files#diff-9147cf18a09e7a07552d042ca6ad847964f68be9279d1956fe8225194ffdb636R348 ?